### ISSUE TITLE
wasm-ld misidentifies library as non-wasm when linking no symbols

### ISSUE THREAD

**Issue by RafaelJBaptista** [llvm/llvm-project#104912]
I built a simple static library, libkx.a. The way my build system works, when you include a library it will make the headers available to the code and automatically add the library to the list of libraries to link.

In my case the libkx.a library is needed just for the headers. When the library is added and the project linked I get a warning:

wasm-ld: warning: ../../../../kst/bin/wasi/debug/libkst.a: archive member 'libkx.a' is neither Wasm object file nor LLVM bitcode

If I unpack the library (llvm-ar -x ../bin/wasi/debug/libkx.a) and inspect each object (llvm-readobj-14 -h) I can confirm that all included objects are in fact WASM, this is typical, but all included objects are ok:

"File: kx_murmur3.o
Format: WASM
Arch: wasm32
AddressSize: 32bit
Version: 0x1
Inspecting kx_mutex_common.o:
"

This same library compiles fine, without errors or warnings when linking it when things in it are actually needed. But in this particular case where  the library is not actually needed I get this spurious warning.

I eliminated the library from the link line - so its solved for me - but this is an incorrect warning and will lead to confusion for many people (as it did for me)

**Comment by sbc100** [comment]
It looks like you have an archive file within another archive file (libkx.a inside of libkst.a). 

I don't think this is support by wasm-ld, or by any other version of lld.  

I tried a simple example with the ELF linker and got a similar error:

```
ld.lld: warning: libhello2.a: archive member 'libhello.a' is neither ET_REL nor LLVM bitcode
```
